src/datasets/argument_relation_detection_dataset.py

```diff
@@ -13,28 +13,34 @@ def __init__(
         self,
         json_path: Path,
         tokenizer: PreTrainedTokenizer,
+        test_mode: bool = False,
     ):
         with open(json_path, "r") as f:
             data = json.load(f)
-        self.texts: list[tuple[str, str]] = [(sample[0], sample[1]) for sample in data]
-        self.labels: list[int] = [sample[2] for sample in data]
+        self.texts: list[tuple[str, str]] = [(sample[1], sample[2]) for sample in data]
+        self.labels: list[int] = [sample[3] for sample in data] if not test_mode else None
         self.tokenizer = tokenizer
+        self.test_mode = test_mode
 
     def __len__(self):
-        return len(self.labels)
+        return len(self.texts)
 
     def __getitem__(self, i):
         texts = self.texts[i]
-        label = self.labels[i]
+        label = self.labels[i] if self.labels is not None else None
```

Comment:
[nitpick] the same thing that is checked by 

if self.labels is not None

here, is checked by

if not self.text_mode

a few lines down. Right now these two things are equivalent, but in a real-world scenario you would still like to use the same "if" statement for both of them, in case some adjustments to the code are made later, and these two things stop being equivalent. This can result in errors that are hard to debug.

In fact, since these things are equivalent, we might want to _not_ have the self.test_mode attribute at all, and always check that by self.labels is not None. Or, we could have a method

def self.test_mode():
  return self.labels is None

which is readable and easy to adjust in the future if that's needed.